ansys-templates edits to doc (#39)
pyansys/ansys-templates: doc/source/user_guide/templating.rst

How to Add a New Template
=========================

> You can easily add new PyAnsys templates. However, before doing so, it is important that

Reviewer: ```suggestion
You can easily add new templates. However, before doing so, it is important that
```

> you read the CONTRIBUTING section.

Author: Is this section yet to be added or will it be to the overall Contributing section in the PyAnsys Developer's Guide. I assume that it will someday have a contributing.rst file of its own that references the Contributing topic in the PyAnsys Developer's Guide. At that point, we should have the link text here be Contributing--not CONTRIBUTING.

Reviewer: Good catch on this one, @PipKat! I was planning to link this to some contributing section when available in the dev-guide. I guess that for the moment we can link this to https://dev.docs.pyansys.com/guidelines/dev_practices.html#contributing-through-github

Reviewer: Linked this to https://github.com/pyansys/dev-guide/issues/64

> you read the CONTRIBUTING section.

Reviewer: ```suggestion
you read the Contributing section.
```

Understanding the Templates Directory
-------------------------------------

> In ``pyansys-templates``, all templates are placed in ``src/ansys/templates/``.

Reviewer: ```suggestion
In ``ansys-templates``, all templates are placed in ``src/ansys/templates/``.
```

          └── {{cookiecutter.__project_name_slug}}/
              └── ... # All these files will be included in each family template

Inside this new family, you must create at least two different files:

- A ``cookiecutter.json`` file for specifying the minimum required variables for
  any of this family's templates to work. You must use cookie cutter private
  variables in this file.

- A ``common/{{cookiecutter.__project_name_slug}}/`` directory to contain all of

> the typical files for this family. All files in the ``common/{{cookiecutter.__project_name_slug}}``
> directory will be copied to each template when rendering it.

Author: ```suggestion
  the typical files for this family. All files in the ``common/{{cookiecutter.__project_name_slug}}``
  directory will be copied to each template when rendering it.
```

          └── {{cookiecutter.__project_name_slug}}/
              └── ... # All these files will be included in each family template

Inside this new family, you must create at least two different files:

- A ``cookiecutter.json`` file for specifying the minimum required variables for
  any of this family's templates to work. You must use cookie cutter private
  variables in this file.

- A ``common/{{cookiecutter.__project_name_slug}}/`` directory to contain all of
  the typical files for this family. All files in the ``common/{{cookiecutter.__project_name_slug}}``
  directory will be copied to each template when rendering it.
  
  
.. note::

    You can later add a ``hooks/post_gen_project.py`` file in the
    ``family/template/`` directory if you need to remove non-desired files coming from
    the ``common/{{cookiecutter.__project_name_slug}}`` directory.

Adding a New Template to a Family
---------------------------------

> To add a new template to a fmaily, first create a new template folder. For example,

Reviewer: ```suggestion
To add a new template to a family, first create a new template folder. For example,
```

Each family of templates is tested within its own `GitHub actions`_ workflow.
Therefore, you need to create a YML file for a new family:

- ``.github/workflows/templates_family.yml``

.. note::

   To reduce the amount of CI jobs, templates are only tested under Linux.
   If you require testing from a particular programming language, try to test the
   minimum and maximum supported versions of the language. Avoid all intermediate
   versions if possible.

Removing Undesired Files
------------------------

It is likely that there are some files coming from the ``common/``
directory that you do not want included in your rendered template. To exclude files,
you can take advantage of `cookiecutter hooks`_. 

Hooks are Python scripts that allow you to control the rendering process both before
and after the process is executed. With hooks, you can move or delete any files
included in the final rendered project.

To use hooks, you must create a new directory named ``src/ansys/templates/new_family/new_template/hooks``.
Only two hooks are allowed:

- ``pre_gen_project.py``: executes before rendering process.
- ``post_gen_project.py``: executes after the rendering process.

.. warning::

>    Both hooks are executed once the cookie cutter context has been started. This

Reviewer: It is the name of the tool, see https://cookiecutter.readthedocs.io/

```suggestion
   Both hooks are executed once the cookiecutter context has been started. This
```
